pkg/didcomm/common/service/destination.go

```diff
@@ -54,10 +64,67 @@ func CreateDestination(didDoc *diddoc.Doc) (*Destination, error) {
 
 	// TODO ensure recipient keys are did:key's
 	//  https://github.com/hyperledger/aries-framework-go/issues/1604
+	// above TODO is TBD - for now added KMS encryption key to support new Authcrypt/Anoncrypt
```

Comment:
a TODO about a TODO....

Reply:
it says TBD of the above TODO.. not sure if we want to support fetching keys from did:key, currently keys are in-lined and encryption keys are found in KeyAgreement that contains a full JsonWebKey2020 JWK key, it could be considered a did:key, not sure we need anything more.. this is why this TBD is added

Reply:
@Baha-sk 

> it says TBD of the above TODO.. not sure if we want to support fetching keys from did:key, currently keys are in-lined and encryption keys are found in KeyAgreement that contains a full JsonWebKey2020 JWK key, it could be considered a did:key, not sure we need anything more.. this is why this TBD is added

Current [DIDComm service conventions](https://github.com/hyperledger/aries-rfcs/blob/master/features/0067-didcomm-diddoc-conventions/README.md#service-conventions) <-- this is what we should be following (for now).

Reply:
@llorllale This convention only states RecipientKeys, which are supposedly representing verification keys. The new packers will use separate keys for packing/unpacking that will be regarded as encryption keys. The way to allow this separation, we need to use the DID Doc's KeyAgreement method to store the encryption key (similar to `X25519KeyAgreementKey2019` shown in [this example](https://w3c.github.io/did-core/#example-20-key-agreement-property-containing-two-verification-methods) but using JSONWebKey2020 as type, pending approval by the w3c community).

This way, we don't diverge from the [DIDComm service conventions](https://github.com/hyperledger/aries-rfcs/blob/master/features/0067-didcomm-diddoc-conventions/README.md#service-conventions).

Reply:
@Baha-sk 

> This convention only states RecipientKeys, which are supposedly representing verification keys.

`recipientKeys` is an array of `did:key` that resolve to DID docs that look like [this](https://w3c-ccg.github.io/did-method-key/#example-2-a-did-document-derived-from-a-did-key) (note the `keyAgreement`). Having the keyAgreement key inlined like this (in the form of did:key references) allows a uniform representation across DIDComm usages, whether it's in a DID Doc or in an invitation (ie. no DID doc).

Please follow the service conventions - let's not reinvent our own unless there is a good reason for it.

Reply:
@llorllale  this doc creation from did:key implies key conversion from ed25519 to X25519. We aim to separate verification from encryption key (ie no conversion or value relationship between the keys). If this is not the goal, then encryption key might as well be the verification key for ECDSA keys. I'm sure this key update is all about increasing security by separating the two. 

If we want separate enc and ver keys, this also means we can't create (expand) a DID doc from did:key. But the example your showing can only work using key conversion (to create the `KeyAgreement` part of the document)

Reply:
@Baha-sk 

> the example your showing can only work using key conversion (to create the KeyAgreement part of the document to contain the converted X25519 key from the ED25519 ver key).

That is exactly how `did:key` works with Ed25519 keys. 

The point "but why not just multicodec+multiformat the key agreement key?" was raised at the time the decision to use `did:key` in DIDComm was made. The answer is pragmatic: to do so, the DIDComm community would have to invest resources into developing the formats and encoding in a proper standards body. It was decided to just use did:key which has done part of the legwork already.

So in the end we need to use did:key in `recipientKeys` and `routingKeys` to remain compliant with Aries DIDComm.

Reply:
this only works for ED25519 keys @llorllale , if we want to use EC verification keys for signature verification and separate (unrelated) EC keys for packing/unpacking then we can't generate a DID doc from `did:key`.

I posted a comment [here](https://github.com/w3c/did-core/issues/185#issuecomment-668658079) to ask the community.

Reply:
@llorllale if I understood this [comment](https://github.com/w3c/did-core/issues/185#issuecomment-668668618) correctly, we will need to add did:key support to find `somehow` the `encryption` keys from the `RecipientKeys` list to do do Pack/Unpack. The remaining keys should be used for verification. 

This means the framework's logic in handling this list should change (not all keys are verification keys in the list and therefore should use only a subset representing verification only keys). 

This is beyond the scope of this patch and can be done in a followup change.

Reply:
You mean after [this](https://github.com/hyperledger/aries-framework-go/pull/2068#discussion_r463957199) comment? :)

Reply:
comment has been removed